crates/checkpoint-downloader/src/config.rs

```diff
@@ -134,11 +134,11 @@ impl Default for AdaptiveDownloaderConfig {
 
 impl AdaptiveDownloaderConfig {
     pub(crate) fn message_queue_size(&self) -> usize {
-        self.max_workers * self.channel_config.work_queue_buffer_factor
+        self.max_workers * self.channel_config.work_queue_buffer_factor * 2
     }
 
     pub(crate) fn checkpoint_queue_size(&self) -> usize {
-        self.max_workers * self.channel_config.work_queue_buffer_factor
+        self.max_workers * self.channel_config.work_queue_buffer_factor * 2
     }
```

Comment:
Question: What is the motivation for these changes?

Reply:
Same as the fix: the channel should have some buffer and shouldn't ever be smaller than application logic limit, otherwise, the unexpected issue like this happens. So essentially, application level sets the soft limit, and underlying channel sets the upper limit, and lower limit < upper limit.